src/bin/basic_subclass.rs

```diff
@@ -53,21 +57,38 @@ impl ObjectImpl for SimpleWindowPrivate {
         let self_ = obj.downcast_ref::<SimpleWindow>().unwrap();
 
         let headerbar = gtk::HeaderBar::new();
+        let increment = gtk::Button::new_with_label("Increment!");
+        let label = gtk::Label::new(Some("Press the Increment Button!"));
+
         headerbar.set_title(Some("Hello World!"));
         headerbar.set_show_close_button(true);
+        headerbar.pack_start(&increment);
 
-        let label = gtk::Label::new(Some("Hello World!"));
+        let weak = self_.downgrade();
```

Comment:
You probably want to use the new `clone!` macro here, or @GuillaumeGomez will complain :)

Reply:
Too late! I'm complaining!

Reply:
How dare you!